add error checking to the foundationdb module and allow persist() to …

zoltan opened the pull request:
…return an error.

this allows us to handle errors in the individual metadata backends better in the future.

zoltan committed:
add error checking to the foundationdb module and allow persist() to return an error

Signed-off-by: Zoltan Arnold Nagy <zoltan.arnold.nagy@gmail.com>

zoltan committed:
Merge branch 'master' into error-handling-fixes

nielspardon reviewed (commented):
shouldn't we check for the return code of chk calls? commenting here since I can't comment on the unchanged lines through the Github UI.

zoltan committed:
make all methods static in the foundationdb module

since the metadata backend modules do not declare an external header
and they should be only used via the appropriate function pointers,
it does not make any sense to have non-static code in the module.

Signed-off-by: Zoltan Arnold Nagy <zoltan.arnold.nagy@gmail.com>

zoltan committed:
check the return values for chk() at multiple places

unfortunately if the the outer function does not return an int
we swallow the return code for now.

Signed-off-by: Zoltan Arnold Nagy <zoltan.arnold.nagy@gmail.com>

zoltan committed:
typo

Signed-off-by: Zoltan Arnold Nagy <zoltan.arnold.nagy@gmail.com>

nielspardon reviewed (commented):
I believe there's one `return proto_message_persist` missing for `physical_disk_range_persist()`

zoltan committed:
return the correct return value from physical_disk_range_persist

Signed-off-by: Zoltan Arnold Nagy <zoltan.arnold.nagy@gmail.com>

nielspardon reviewed (approved):
LGTM 🎉
